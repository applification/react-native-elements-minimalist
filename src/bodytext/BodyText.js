import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-primitives';

class HeaderTextElement extends Component {
  static propTypes = {
    color: PropTypes.string.isRequired,
    children: PropTypes.any,
  };
  static defaultProps = {
    color: '#333333',
  };

  render() {
    return (
      <Text
        style={{
          color: this.props.color,
          paddingLeft: 25,
          paddingRight: 25,
          paddingBottom: 15,
          fontWeight: 'normal',
          fontSize: 17,
        }}
      >
        {this.props.children}
      </Text>
    );
  }
}

export default HeaderTextElement;
