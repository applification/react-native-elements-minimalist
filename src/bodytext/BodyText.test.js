import React from 'react';
import renderer from 'react-test-renderer';

import BodyText from './BodyText';

describe('HEADER TEXT', () => {
  it('renders BodyText component', () => {
    expect(renderer.create(<BodyText>Body text</BodyText>)).toMatchSnapshot();
  });

  it('renders BodyText in red using passed in prop', () => {
    expect(
      renderer.create(<BodyText color="red">Red text</BodyText>)
    ).toMatchSnapshot();
  });

  it('renders BodyText with a fontsize of 25, normal', () => {
    const componentOutput = JSON.parse(
      JSON.stringify(renderer.create(<BodyText>Header 50px</BodyText>))
    );
    const fontSize = componentOutput.props.style.fontSize;
    const fontWeight = componentOutput.props.style.fontWeight;

    expect(fontSize).toBe(17);
    expect(fontWeight).toBe('normal');
  });
});
