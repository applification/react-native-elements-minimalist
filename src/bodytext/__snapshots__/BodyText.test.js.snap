// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HEADER TEXT renders BodyText component 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#000000",
      "fontSize": 17,
      "fontWeight": "normal",
      "paddingBottom": 15,
    }
  }
>
  Body text
</Text>
`;

exports[`HEADER TEXT renders BodyText in red using passed in prop 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "red",
      "fontSize": 17,
      "fontWeight": "normal",
      "paddingBottom": 15,
    }
  }
>
  Red text
</Text>
`;
